addons/website_blog/views/website_blog_components.xml

```diff
@@ -21,9 +22,48 @@
                     </a>
                 </li>
             </ul>
+            <!-- Mobile -->
+            <t t-set="navIsOffCanvas" t-value="len(blogs) > 4"/>
+            <div t-if="len(blogs) > 1" t-attf-class="order-2 d-lg-none #{not navIsOffCanvas and 'dropdown flex-grow-1 flex-sm-grow-0 min-w-0'}">
+                <a t-attf-class="btn btn-light #{not navIsOffCanvas and 'dropdown-toggle d-flex align-items-center justify-content-between'}" 
+                   type="button" 
+                   role="button" 
+                   t-attf-title="{{blog.name if blog else 'All Blogs'}}" 
+                   t-att-data-bs-toggle="navIsOffCanvas and 'offcanvas' or 'dropdown'" 
+                   t-att-data-bs-target="navIsOffCanvas and '#o_wblog_offcanvas'" 
+                   t-att-aria-expanded="not navIsOffCanvas and 'false'" 
+                   t-att-aria-controls="navIsOffCanvas and 'o_wblogs_offcanvas'">
+                    <i t-if="navIsOffCanvas" class="fa fa-navicon"/>
+                    <t t-else="">
+                        <span t-if="blog" class="d-inline-block text-truncate" t-out="blog.name"/>
+                        <span t-else="" class="d-inline-block text-truncate">All</span>
+                    </t>
+                </a>
+                <div id="o_wblog_offcanvas" t-att-class="navIsOffCanvas and 'offcanvas offcanvas-end d-lg-none' or 'dropdown-menu dropdown-menu-end'">
+                    <div t-if="navIsOffCanvas" class="offcanvas-header">
+                        <h5 class="offcanvas-title">Blogs</h5>
```

Comment:
Unfortunately, starting from `17.0` we have to add `mt-0` and `mb-[x]` on any **UI** headings 'cause these are affected by user customizations 🙄